pegasus-spark-bulkloader/src/main/scala/com/xiaomi/infra/pegasus/spark/bulkloader/PegasusRecord.scala

```diff
@@ -0,0 +1,106 @@
+package com.xiaomi.infra.pegasus.spark.bulkloader
+
+import java.nio.{ByteBuffer, ByteOrder}
+import java.util.Date
+
+import com.google.common.primitives.Bytes
+import org.apache.commons.lang3.Validate
+import org.apache.commons.lang3.builder.HashCodeBuilder
+
+object PegasusRecord {
+  private val EPOCH_BEGIN = 1451606400 // seconds since 2016.01.01-00:00:00 GMT
+
+  def generateKey(hashKey: Array[Byte], sortKey: Array[Byte]): Array[Byte] = {
+    val hashKeyLen =
+      if (hashKey == null) 0
+      else hashKey.length
+    Validate.isTrue(
+      hashKeyLen < 65535,
+      "length of hash key must be less than UINT16_MAX",
+      new Array[AnyRef](0)
+    )
+    val sortKeyLen =
+      if (sortKey == null) 0
+      else sortKey.length
+    val buf = ByteBuffer.allocate(2 + hashKeyLen + sortKeyLen)
+    buf.putShort(hashKeyLen.toShort)
+    if (hashKeyLen > 0) buf.put(hashKey)
+
+    if (sortKeyLen > 0) buf.put(sortKey)
+
+    buf.array
+  }
+
+  // todo(jiashuo1): ttl, ts, clusterId, deleteTag default 0, later offer api to set it
+  def generateValue(
+      value: Array[Byte],
+      ttl: Int = 0,
+      ts: Long = 0,
+      clusterId: Short = 0,
+      deleteTag: Byte = 0
```

Comment:
deleteTag is a `boolean`

Reply:
It can't be "|" when I set `boolean`,  So I replace it `Byte`

Reply:
I have changed : `(if (deleteTag) Byte(1) else Byte(0)))`